### DISCUSSION TITLE
TypeError: maybeGlobalAsyncLocalStorage.snapshot is not a function in NextJS 15

### DISCUSSION THREAD

**Discussion by reydoblas** [vercel/next.js#72074] [Help]
### Summary

I am experiencing TypeError: maybeGlobalAsyncLocalStorage.snapshot is not a function when running `npm run dev` or `npm run build` when upgrading to NextJS 15. Below are my setup info,

OS: Ubuntu 24.04 LTS
Node: v23.1.0
npm: 10.9.0
React: 18.3.1

Anyone experiencing same issue? 

Thanks.
![2024-10-30_17-52](https://github.com/user-attachments/assets/398a547c-e58c-4b1a-b930-de6678e3483f)

**Comment by icyJoseph** [comment]
Could you try downgrading Node to LTS?

**Reply by reydoblas** [reply]
I've already tried downgrading Node to versions 22.11.0, 20.18.0, and 18.20.4 but still same error.

**Reply by icyJoseph** [reply]
Could it be that a client component is accessing headers/cookies or any other kind of API available server side only?

**Comment by reydoblas** [comment]
I don't think that is the case since everything works fine in Windows and
MAC OS. The problem only occurs in my Ubuntu setup.

**Reply by icyJoseph** [reply]
Aha, that complicates things... So somehow Node.js' async hooks are not working properly on that distribution... 🤔 Or at least Next.js' implementation can't find the async local storage... are you able to tell which part of the app causes this?

**Reply by icyJoseph** [reply]
I wonder if you delete the lock file, and reinstall all dependencies?

**Comment by reydoblas** [comment]
Done that already but still no luck. Tried both npm install and update
command.

**Reply by icyJoseph** [reply]
🤔 ~ I haven't been able to find any Ubuntu related issues either 

I don't have a machine with that distribution, sorry I can't help further.

Have you tried running with the Node debugger, set to break on exceptions? Perhaps that can give us some more information?

**Comment by reydoblas** [comment]
I haven't tried using the Node debugger. I will try to dig into this
further and get back here if I have any findings or solution. Thanks for
the help.

**Comment by jakobsuckow** [comment]
I am having the same issue.

**Comment by robertfrontend** [comment]
I have the same problem.

**Comment by icyJoseph** [comment]
Anyone with a minimal reproduction repository?

**Comment by jtmuller5** [comment]
I downgraded to version [14.2.6](https://www.npmjs.com/package/next/v/14.2.16) and it started working again.

**Reply by icyJoseph** [reply]
From what version? Are you able to binary search (or linearly) the first version that breaks?

**Comment by mrtnrs** [comment]
I just set up the vercel [nextjs-postgres-auth-starter](https://github.com/vercel/nextjs-postgres-auth-starter) clean (MAC), got this error. Downgrading next to 14.2.14 resolved it.

**Comment by nathekame** [comment]
I am using Ubuntu 22.04 I encountered the same problem but when i downgraded to Next 14.2.6 and  everything works fine now

**Comment by icyJoseph** [comment]
I think, for most, the error is the Node.js version, as per https://github.com/vercel/next.js/issues/72077 this issue and its docs fix... 

For everyone else, who has the correct range of Node, are you using next-auth as well? 

Are you 100% sure that the Node you are running is in the correct range? The next command might sometimes find a different node in your system. Try to log `process.version` on your server side code.
